hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/JpaArtifactManagement.java

```diff
@@ -125,11 +121,16 @@ public Artifact create(final ArtifactUpload artifactUpload) {
     }
 
     private AbstractDbArtifact storeArtifact(final ArtifactUpload artifactUpload) {
-        try {
-            return artifactRepository.store(tenantAware.getCurrentTenant(), artifactUpload.getInputStream(),
+        AbstractDbArtifact dbArtifact = null;
+        try(final InputStream quotaStream = wrapInQuotaStream(artifactUpload.getInputStream())) {
+
+            dbArtifact = artifactRepository.store(tenantAware.getCurrentTenant(), quotaStream,
```

Comment:
Could you directly return the result here, instead of saving it in `dbArtifact`, because you don't use it later.

Reply:
sure, is a left over of intermediate fixes ;)